mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerWithEngine.java

```diff
@@ -296,6 +300,45 @@ public void testInsert() throws IOException {
     HiveIcebergTestUtils.validateData(table, new ArrayList<>(HiveIcebergStorageHandlerTestUtils.CUSTOMER_RECORDS), 0);
   }
 
+  @Test
+  public void testInsertSupportedTypes() throws IOException {
+    Assume.assumeTrue("Tez write is not implemented yet", executionEngine.equals("mr"));
+    for (int i = 0; i < SUPPORTED_TYPES.size(); i++) {
+      Type type = SUPPORTED_TYPES.get(i);
+      // TODO: remove this filter when issue #1881 is resolved
+      if (type == Types.UUIDType.get() && fileFormat == FileFormat.PARQUET) {
+        continue;
+      }
+      // TODO: remove this filter when we figure out how we could test binary types
+      if (type.equals(Types.BinaryType.get()) || type.equals(Types.FixedType.ofLength(11))) {
+        continue;
+      }
+      String tableName = type.typeId().toString().toLowerCase() + "_table_" + i;
+      String columnName = type.typeId().toString().toLowerCase() + "_column";
+
+      Schema schema = new Schema(required(1, "id", Types.LongType.get()), required(2, columnName, type));
+      List<Record> expected = TestHelper.generateRandomRecords(schema, 5, 0L);
+      List<Record> records = new ArrayList<>();
+      if (type == Types.TimestampType.withoutZone()) {
+        expected.forEach(r -> records.add(r.copy()));
+        records.forEach(r -> r.set(1, Timestamp.valueOf((LocalDateTime) r.get(1))));
+      } else if (type == Types.TimestampType.withZone()) {
+        expected.forEach(r -> records.add(r.copy()));
+        records.forEach(r -> r.set(1, Timestamp.from(((OffsetDateTime) r.get(1)).toInstant())));
+      } else {
+        records.addAll(expected);
+      }
+
+      Table table = testTables.createTable(shell, tableName, schema, fileFormat, ImmutableList.of());
+      StringBuilder query = new StringBuilder("INSERT INTO ").append(tableName).append(" VALUES")
+              .append(String.join(",", records.stream().map(r ->
+                  String.format(type == Types.BooleanType.get() ? "(%s,%s)" : "(%s,'%s')", r.get(0),
+                      r.get(1))).collect(Collectors.toList())));
```

Comment:
I think you can use `Collectors.joining` here and then you wouldn't need the extra `String.join` step

Reply:
Good point! Done.